docs/sources/flow/concepts/clustering.md

```diff
@@ -0,0 +1,243 @@
+---
+title: Clustering
+weight: 500
+labels:
+  stage: beta
+---
+
+# Clustering (beta)
+
+Clustering enables Grafana Agent Flow to coordinate a fleet of agents working
+together for workload distribution and high availability. It helps create
+horizontally scalable deployments with minimal resource and operational
+overhead.
+
+To achieve this, Grafana Agent makes use of an eventually consistent model that
+assumes all participating Agents are interchangeable and converge on using the
+same configuration file.
+
+The behavior of a standalone, non-clustered agent is the same as if it was a
+single-node cluster.
+
+In comparison to [hashmod sharding][] as a scaling mechanism, clustering with
+target auto-distribution provides high availability without requiring multiple
+replicas for the same shard, and can dynamically reshard targets without
+changes to the configuration file.
+
+## Use cases
+
+[Setting up][] clustering using the command-line arguments is the first step in
+making agents aware of one another.
+Components in a telemetry pipeline need to explicitly opt-in to participating
+to a clustering use case in their River config.
+
+### Target auto-distribution
+
+Target auto-distribution is the most basic use case of clustering; it allows
+scraping components running on all peers to distribute scrape load between
+themselves. All nodes must have access to the same service discovery APIs, and
+the set of targets should converge on a timeline comparable to the scrape
+interval.
+
+Whenever a cluster state change is detected, either due to a new node joining
+or an existing node going away, all participating components locally
+recalculate target ownership and rebalance the number of targets they’re
+scraping without explicitly communicating ownership over the network.
+
+The agent makes use of a fully-local consistent hashing algorithm to distribute
+targets, meaning that on average only ~1/N of the targets are redistributed.
+This is in contrast to hashmod sharding where up to 100% of the targets could
+be reassigned to another node and possibly cause system instability.
```

Comment:
Ditto here with the comparison: I'm not sure if we should be trying to sell people on clustering compared to other approaches here, and instead we could focus solely on the benefits of clustering (esp. since it's the only scaling mechanism we're choosing to document for Flow mode).